### DISCUSSION TITLE
a P2P exchange for Sunflower assets

### DISCUSSION THREAD

**Discussion by S3mz** [sunflower-land/sunflower-land#1160] [Show and tell]
## Sunflower Land assets can now be traded safely by community members

Gm fam 👋

Some of you may already know us and about the project but we're happy to share with you that the [Web3 Bazaar](https://web3bazaar.org) is now live on Polygon Network and <b>we have already listed and verified all Sunflower Land assets to enable our community to safely trade tokens and NFTs in batches or individually with other members</b>.

### here's an SFL trade we already made in the Bazaar:
>Check it on Polygonscan [here](https://polygonscan.com/tx/0x7a4540bd87de19a355a44c7595f8da3fc6f1d2cf96d12f1d68794bfa22a1f912)

One nice little Bunny for a useful Beaver and some SFL : <br>
<img width="545" alt="td1" src="https://user-images.githubusercontent.com/92461670/176669280-b947d1df-4aad-435a-be77-471a034074cd.png">
<img width="545" alt="td2" src="https://user-images.githubusercontent.com/92461670/176669365-f847c65e-cae6-4e9f-977c-2e1d0771596d.png">
<img width="545" alt="td3" src="https://user-images.githubusercontent.com/92461670/176669909-064b55af-325f-4b9b-8f35-7eb145c98bf5.png">

### Bazaar and SFL. Let's partner up?

To celebrate our Mainnet launch we'd love to scoop some rare SFL NFTs and giveaway to the community through the Bazaar and start showing the world there's a now a fully permissionless non-custodial way of doing P2P trades.

### About the Web3 Bazaar

Web3 Bazaar is a dApp is supported by non-custodial escrow contracts that enable peer-to-peer swaps of ERC-20, ERC-721 or ERC-1155 tokens. Contracts' only purpose is to switch asset ownership from one wallet address to another under the trades pre-established in it.

#### Status of the Project

- [Smart contracts](https://polygonscan.com/address/0x93cdc98317A07e83a9AA96F69AdA7Af4b37EBf44) live on Polygon network
- Community-tested, audit undergoing (by Certik)
- Polygon DAO grantee
- [Gitcoin Round 14](https://gitcoin.co/grants/5653/web3-bazaar) top 60 projects in number of contributor

**Comment by MoonLabsDev** [comment]
Sounds nice, actually we have something similar aswell already and it is used on _High Seas Piracy_ Tradeserver.

**Functions:**
- Batch Trading
- Dynamic offer system
- 100% onchain
- Trade SFL Items, MATIC or a range of whitelisted tokens and ofcourse SFL
- Overview over errors (insufficient balance, missing approval,..)
- History of all executed trades
- setting up a receiver wallet per trade, so that you can directly receive your items in your farm

![2022-07-01 02_19_41-Window](https://user-images.githubusercontent.com/103362189/176801260-0d4eb54b-36d4-4bcb-93c2-fd534070bcaf.png)
![2022-07-01 02_20_20-Window](https://user-images.githubusercontent.com/103362189/176801266-11273c54-e04c-4c45-8820-86e5f2a19613.png)
![2022-07-01 02_20_32-Window](https://user-images.githubusercontent.com/103362189/176801276-6dff07af-4fa4-4217-948b-3c8c197d62d6.png)
![2022-07-01 02_21_08-Window](https://user-images.githubusercontent.com/103362189/176801280-84b9d47f-25b8-4038-8a75-5f3bb8e41ff9.png)
![2022-07-01 02_32_28-Window](https://user-images.githubusercontent.com/103362189/176801283-cc5e5170-d889-4284-b4ed-c72f314cf600.png)

Version is live at https://v2.moon-vault.com
We have KYC aswell and audit will follow soon. Also we have a Dex Aggregator onsite, so that you can directly swap your tokens to whatever your need.

![2022-07-01 02_46_51-Window](https://user-images.githubusercontent.com/103362189/176801288-b922e80c-5fa1-4d9f-ba74-e5b67ecd6c78.png)

**Comment by S3mz** [comment]
yep cool one.

We Just don't do KYC and will never do (the exchange is fully permiossonless),  We don't charge any fee ever (this is a community project and for the community) and sure, the Bazaar also supports batch transfers, all SFL assets (and actually all other Polygon and EVM chains assets) and is non-custodial (meaning that users' tokens never leave their wallet until the moment of swap)
